### PR TITLE
Add support for SUBMIT_SLEEP

### PR DESCRIPTION
**Pull request by berland**
Written assuming there will be a global option to set SUBMIT_SLEEP, but will currently pick from the selected queue system.

### CODE REVIEW

File: src/ert/job_queue/queue.py [hunk 1]
```diff
@@ -159,6 +159,12 @@ def count_running(self) -> int:
             for real in self._realizations
         )
 
+    def count_submitted(self) -> int:
```
Comment (xjules) [inline src/ert/job_queue/queue.py:162]: Now you can do `self.count_realization_state(RealizationState.SUBMITTED)`

**Comment by xjules** [comment]
I think you need to rebase

**Comment by berland** [comment]
Should be ready now, with tests

File: src/ert/job_queue/queue.py [hunk 2]
```diff
@@ -166,27 +169,37 @@ def max_running(self) -> int:
...
+        time_of_last_submission: datetime.datetime = (
+            datetime.datetime.now() - submit_delta
+        )
+        while self.is_active():
+            while (
+                self.count_realization_state(RealizationState.RUNNING)
+                + self.count_realization_state(RealizationState.SUBMITTED)
+                < self.max_running()
+            ):
+                if datetime.datetime.now() - time_of_last_submission >= submit_delta:
```
Comment (xjules) [inline src/ert/job_queue/queue.py:186]: I thought that this check is not required if we do `await asyncio.sleep(self._queue_config.submit_sleep)` or?

Comment (berland) [reply src/ert/job_queue/queue.py]: removed now

**Review by JHolba — APPROVED** [review]

**Pull request merged**